conf/distro/rpb-wayland.conf

```diff
@@ -11,3 +11,7 @@ DISTRO_FEATURES_remove = "x11"
 PREFERRED_PROVIDER_virtual/egl_hikey = "mali450-userland"
```

Comment:
Instead of adding a dozen PREFERRED_PROVIDER_virtual/egl_ entries for each and every board using mali, use an anonymous python method to set it, based on MACHINE_FEATURE, along the lines of:

```
def mali_handler(d):
    features = d.getVar('MACHINE_FEATURES', True).split()
    if 'mali' in features:
        egl = 'mali-userland'
    else:
        egl = d.getVar('PREFERRED_PROVIDER_virtual/egl', True)
    return egl;

PREFERRED_PROVIDER_virtual/egl := "${@mali_handler(d)}"
```

Someone with actual python knowledge can improve the above, it's just to illustrate the idea.